ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js

```diff
@@ -170,6 +176,37 @@ Query.Model = Backbone.AssociatedModel.extend({
     this.handleChangeResultCount()
     this.startSearch(options)
   },
+  startTieredSearch: function(ids) {
+    this.set('federation', 'local')
+    $.when(...this.startSearch()).then(
+      function() {
+        setTimeout(() => {
+          const results = this.get('result')
+            .get('results')
+            .toJSON()
+
+          const status = this.get('result')
+            .get('status')
+            .toJSON()
+
+          const resultIds = results.map(result => result.metacard.id)
+          const missingResult = ids.some(id => !resultIds.includes(id))
+          if (!missingResult) {
+            return
+          }
+          this.set('federation', 'enterprise')
+          this.startSearch()
+          this.get('result')
```

Comment:
Interesting 🤔 

Might be more more readable if we do this (setting the results and status) by passing in options to startSearch.